magma/backend/mlir/mlir_passes.py

```diff
@@ -0,0 +1,25 @@
+import abc
+
+from magma.backend.mlir.mlir import MlirOp
+
+
+class MlirOpPass:
+    def __init__(self, root: MlirOp):
+        self._root = root
+        self._callable = callable(self)
+
+    def _run_on_op(self, op: MlirOp):
+        for region in op.regions:
+            for block in region.blocks:
+                for op in block.operations:
+                    yield from self._run_on_op(op)
+                    if self._callable:
+                        yield self(op)
```

Comment:
Would this be better as an abstact base class with __call__ required to be defined? Otherwise this logic seems not useful (runs the traversal but doesn't do anything with the traversed elements)

Reply:
:+1: